### DISCUSSION TITLE
配合passwall2,用的,  请教下!!!!

### DISCUSSION THREAD

**Discussion by jt8823** [IrineSistiana/mosdns#805] [Q&A]
第一次用mosdns,   看见里面有个规则列表
但是passwall2也有规则列表,
那么使用的时候,   是用的哪个规则呢

**Comment by Sereinfy** [comment]
mosdns的规则列表是用来dns解析分流的
swall规则列表列表 是mosdns发起成功后，建立连接时，用规则来分流流量走那条路的。
一般是先swall分流，再mosdns发起dns。
也可以用dns分流，先发起 再分路

**Reply by jt8823** [reply]
passwall 已经把DNS指向了modns,     那么modns,知道passwall的规则?   在进行分流,  是这样理解吗?

**Reply by Sereinfy** [reply]
举个栗子吧，再不理解没办法了。
比如百度.com。passwall看了看自己的规则，走直连规则。然后把请求给mosdns，mosdns是发起dns用的，dns有很多服务商，看了看自己的规则。百度.com走 1.1.1.1这个dns查询。你用mosdns，可以不让百度1.1.1.1，可以让他走8.8.8.8。这是mosdns规则的作用。不过这是最简单的情况，实际上代理服务器还会发起自己的dns查询

**Reply by jt8823** [reply]
哈哈 对不起   我太笨了   谢谢!!   
是这样吗?   passwall是 分流域名 IP规则的,    mosdns是分流DNS规则的,        那是不是 比如    我想让1.1.1.1解析百度.com   就得在mosdns中的规则列表进行编辑?       不然mosdns也不知道走哪个规则呀,        passwall已有的规则域名ip,   也需要在mosdns的规则列表中进行设置?    不然dns解析,规则还是乱的吧,   还是说passwall   能告诉mosdns   是走哪个DNS?

**Reply by Sereinfy** [reply]
他两个在默认情况下使用的都是 geoip的查询判断。不会冲突。自己知道该怎么走，你操心过头了。不用设置。你面前有两条路，passwall是告诉你走一条路的东西，选了左边的路之后。需要有人给你把拦路的打开（不是很恰当）。这个人可以是任何人，就是dns服务商。路和开锁那个人，是两回事。你这dns都没搞懂，分流也没搞懂。走哪个dns是mosdns告诉的。建议还是把不良林的视频从头到尾看三五遍。我感觉解释的够清楚了 还是没懂

**Reply by jt8823** [reply]
好吧。  谢谢了！

**Comment by iceyheart** [comment]
mosdns先过一次规则，分流到pw再过一次规则
